### DISCUSSION TITLE
TID Problem

### DISCUSSION THREAD

**Discussion by moatms** [FeynCalc/feyncalc#170] [Q&A] [answered]
Hello,
I am using FeynCalc to calculate the amplitude of the box diagrams of gg -> ZZ and I have a problem with TID. My code is:

tops = CreateTopologies[1, 2 -> 2, Adjacencies -> {3}, ExcludeTopologies -> {Tadpoles, Triangles, WFCorrections}];
diags = InsertFields[tops, {V[5], V[5]} -> {V[2], V[2]}, ExcludeParticles -> {S, F[4], F[3, {1}], F[3, {2}]}, InsertionLevel -> {Particles}, Model -> "SMQCD"];
AmpFA = CreateFeynAmp[diags, Truncated -> False];
AmpFA = AmpFA /. PolarizationVector -> FAPolarizationVector;
AmpFC = FCFAConvert[AmpFA, IncomingMomenta -> {p1, p2}, OutgoingMomenta -> {p3, p4}, LoopMomenta -> {q}, List -> False, ChangeDimension -> D, SMP -> True, Contract -> True, TransversePolarizationVectors -> {p1, p2}, DropSumOver -> True, UndoChiralSplittings -> True] // FCTraceFactor // SUNSimplify //PropagatorDenominatorExplicit // Simplify
AmpFC2 = AmpFC /. SMP["m_t"] -> 0 // DiracSimplify
AmpFC3 = AmpFC2 // TID[#, q, ToPaVe -> True] &

The result of AmpFC3 (and so of TID) is 0. 
However it should not be 0: I checked it with some papers and MadGraph and they give non-zero results for this amplitude and so I don't know what to do. I also tried with the mass of the top being different from 0 and I still get that AmpFC3=0. Thanks a lot

**Comment by vsht** [comment, marked as answer]
Hi,

you need to remove `PropagatorDenominatorExplicit`. This function can be applied only to denominators
that are free of loop momenta. Otherwise, rewriting say `FAD[{q,m}]` as `1/(SPD[q]-m^2)` prevents `TID` from
recognizing it as a proper loop integral. So `TID` thinks that your amplitude has loop momenta only in the numerators
and hence regards all integrals as vanishing.

**Reply by moatms** [reply]
Thanks a lot!
